storage/storage-service/src/lib.rs

```diff
@@ -35,6 +35,102 @@ use storage_proto::proto::storage::{
 };
 use tokio::runtime::Runtime;
 
+use libra_secure_net::NetworkServer;
+use libra_types::{account_state_blob::AccountStateBlob, proof::SparseMerkleProof};
+use std::thread::{self, JoinHandle};
+use storage_proto::StartupInfo;
+
+/// Starts storage service with a given LibraDB
+pub fn start_storage_service_with_db_simple(
+    config: &NodeConfig,
+    libra_db: Arc<LibraDB>,
+) -> JoinHandle<()> {
+    let storage_service = StorageServiceSimple { db: libra_db };
+    start_storage_service_loop(config, storage_service)
+}
+
+/// Starts storage service according to config.
+pub fn start_storage_service_new_simple(config: &NodeConfig) -> JoinHandle<()> {
+    let storage_service = StorageServiceSimple::new(&config.storage.dir());
+    start_storage_service_loop(config, storage_service)
+}
+
+fn process_one_message(
+    network_server: &mut NetworkServer,
+    service: &StorageServiceSimple,
+) -> Result<(), Error> {
+    let request = network_server.read()?;
+    let response = service.handle_message(request)?;
+    network_server.write(&response)?;
+    Ok(())
+}
+
+fn start_storage_service_loop(
+    config: &NodeConfig,
+    service: StorageServiceSimple,
+) -> JoinHandle<()> {
+    let mut network_server = NetworkServer::new(config.storage.address);
+    thread::spawn(move || loop {
+        if let Err(e) = process_one_message(&mut network_server, &service) {
+            warn!("Warning: Failed to process message: {}", e);
+        }
+    })
+}
+
+#[derive(Clone)]
+pub struct StorageServiceSimple {
+    db: Arc<LibraDB>,
+}
+
+impl StorageServiceSimple {
+    /// This opens a [`LibraDB`] at `path` and returns a [`StorageService`] instance serving it.
+    pub fn new<P: AsRef<Path>>(path: &P) -> Self {
+        let db = Arc::new(LibraDB::new(path));
+        Self { db }
+    }
+
+    fn handle_message(&self, input_message: Vec<u8>) -> Result<Vec<u8>, Error> {
+        let input = lcs::from_bytes(&input_message)?;
+        let output = match input {
+            storage_interface::StorageMsg::GetAccountStateWithProofByVersionRequest(req) => {
+                lcs::to_bytes(&self.get_account_state_with_proof_by_version(&req))
+            }
+            storage_interface::StorageMsg::GetStartupInfoRequest => {
+                lcs::to_bytes(&self.get_startup_info())
+            }
+            storage_interface::StorageMsg::SaveTransactionsRequest(req) => {
+                lcs::to_bytes(&self.save_transactions(&req))
+            }
+        };
+        Ok(output?)
+    }
+
+    fn get_account_state_with_proof_by_version(
+        &self,
+        req: &storage_interface::GetAccountStateWithProofByVersionRequest,
+    ) -> Result<(Option<AccountStateBlob>, SparseMerkleProof), Error> {
+        Ok(self
+            .db
+            .get_account_state_with_proof_by_version(req.address, req.version)?)
+    }
+
+    fn get_startup_info(&self) -> Result<Option<StartupInfo>, Error> {
```

Comment:
Every time I see startup info, I feel sad.

Reply:
I will move it eventually. You have my word.